lib/sinon/collection.js

```diff
@@ -46,7 +46,16 @@ var collection = {
     },
 
     resetHistory: function resetHistory() {
-        each(this, "resetHistory");
+        getFakes(this).forEach(function (fake) {
+            if (typeof fake.resetHistory === "function") {
+                fake.resetHistory();
+                return;
```

Comment:
No thanks, [there is no else if](http://rmurphey.com/blog/2012/12/10/js-conditionals)